### PR TITLE
Alloy v2

### PR DESCRIPTION
**Pull request by joelrebel**
Implement support to run Alloy as a controller on the JS

depends on
- https://github.com/metal-toolbox/mctl/pull/37
- https://github.com/metal-toolbox/alloy/pull/154

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,8 @@ local-devel-upgrade: kubectl-ctx-kind
 ## uninstall helm chart
 uninstall-local-devel: kubectl-ctx-kind
 	helm uninstall hollow-sandbox
+	# incase the crdb pvc is stuck in terminating
+	# kubectl patch pvc db -p '{"metadata":{"finalizers":null}}'
```
Comment (DoctorVin) [inline Makefile:17]: maybe this should be a different target? `make force-uninstall`?

File: scripts/nats-bootstrap/functions.sh [hunk 2]
```diff
@@ -12,7 +12,9 @@ function clean_natsserver() {
 }
 
 function init_natsserver() {
+	kubectl get pods
```
Comment (DoctorVin) [inline scripts/nats-bootstrap/functions.sh:15]: with out a `diff` it's hard to understand why we do 2 `get pods`?

File: templates/conditionorc-configmap.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ data:
     store_kind: serverservice
     # conditions accepted by this API
     conditions:
-      - kind: inventoryOutofband
+      - kind: inventory
```
Comment (DoctorVin) [inline templates/conditionorc-configmap.yaml:13]: are we scrapping in-band/out-of-band? 🎉

**Review by DoctorVin — APPROVED** [review]
Looks fine. My biggest comment/desire for change is coming up with something better than just repeated `kubectl get pods` to validate that the controllers were deployed properly.

File: Makefile (hunk 1 above)
Comment (joelrebel) [reply Makefile:17]: or we could go with `make install` and  `make clean` ?

File: templates/conditionorc-configmap.yaml (hunk 3 above)
Comment (joelrebel) [reply templates/conditionorc-configmap.yaml:13]: yep, the method of collection would be a parameter in the Condition payload 
https://github.com/metal-toolbox/mctl/blob/41f22410af443bab22cb36a48d2ba2df37447b44/cmd/collect/inventory.go#L48

File: scripts/nats-bootstrap/functions.sh (hunk 2 above)
Comment (joelrebel) [reply scripts/nats-bootstrap/functions.sh:15]: Oh, that is mainly to just eyeball before and after the helm deployment when debugging

File: Makefile (hunk 1 above)
Comment (joelrebel) [reply Makefile:19]: This is now `make install` `make upgrade` and `make clean`

**Review by DoctorVin — DISMISSED** [review]

File: notes/cheatsheet.md [hunk 4]
```diff
@@ -26,3 +26,15 @@ Consume an event as conditionorc
 ```
 nats consumer sub --creds=/root/nsc/nkeys/creds/KO/controllers/conditionorc.creds -s nats://nats:4222
 ```
+
+
+list keys in statusKV buckets
```
Comment (DoctorVin) [inline notes/cheatsheet.md:31]: It's probably more useful to get the keys and examine individual data with `kv get --raw`

**Pull request merged**